ops/scripts/tf_init.sh

```diff
@@ -67,6 +90,33 @@ if [ -z $AWS_REGION ];then
   read -p "You need to specify an AWS region: " AWS_REGION
 fi
 
+# Output values and ask for confirmation
+
+echo ""
+echo "Current directory: $(pwd)"
+echo "Project: $gitlab_project"
+echo "Environment: $target_environment"
+echo "Region: $AWS_REGION"
+echo "GitLab User: $GITLAB_USERNAME"
+echo "Web EC2 Type: $web_ec2_type"
+echo "Bastion EC2 Type: $bastion_ec2_type"
+echo "RDS EC2 Type: $rds_ec2_type"
+echo ""
+
+read -p "Do you want to continue (y/n)?" choice
+case "$choice" in 
+  y|Y ) 
+    echo "yes"
+    ;;
+  n|N ) 
+    echo "no"
+    exit 0
+    ;;
+  * ) 
+    echo "invalid"
+    ;;
```

Comment:
Hi @rija,

In this case, if I input non y or non n, the `tf_init.sh` would still proceed:
```
% ./../../scripts/tf_init.sh --project gigascience/forks/kencho-gigadb-website --env staging
You need to specify an AWS region: ap-northeast-2

Current directory: /Volumes/kencho/gigadb-website/ops/infrastructure/envs/staging
Project: gigascience/forks/kencho-gigadb-website
Environment: staging
Region: ap-northeast-2
GitLab User: kencho
Web EC2 Type: t3.micro
Bastion EC2 Type: t3.micro
RDS EC2 Type: t3.micro

Do you want to continue (y/n)?
invalid

Initializing the backend...
Initializing modules...

Initializing provider plugins...
- Reusing previous version of hashicorp/aws from the dependency lock file
- Reusing previous version of hashicorp/random from the dependency lock file
- Reusing previous version of hashicorp/external from the dependency lock file
- Using previously-installed hashicorp/aws v5.5.0
- Using previously-installed hashicorp/random v3.5.1
- Using previously-installed hashicorp/external v2.3.1

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```

Would it be more reasonable to `exit` as update the code like this:
```
* ) 
    echo "invalid input"
    exit 0
    ;;
``` 
Therefore, for the case of non y and non n case is better addressed.

Reply:
> Hi @rija,
> 
> In this case, if I input non y or non n, the `tf_init.sh` would still proceed:
> 
> ```
> % ./../../scripts/tf_init.sh --project gigascience/forks/kencho-gigadb-website --env staging
> You need to specify an AWS region: ap-northeast-2
> 
> Current directory: /Volumes/kencho/gigadb-website/ops/infrastructure/envs/staging
> Project: gigascience/forks/kencho-gigadb-website
> Environment: staging
> Region: ap-northeast-2
> GitLab User: kencho
> Web EC2 Type: t3.micro
> Bastion EC2 Type: t3.micro
> RDS EC2 Type: t3.micro
> 
> Do you want to continue (y/n)?
> invalid
> 
> Initializing the backend...
> Initializing modules...
> 
> Initializing provider plugins...
> - Reusing previous version of hashicorp/aws from the dependency lock file
> - Reusing previous version of hashicorp/random from the dependency lock file
> - Reusing previous version of hashicorp/external from the dependency lock file
> - Using previously-installed hashicorp/aws v5.5.0
> - Using previously-installed hashicorp/random v3.5.1
> - Using previously-installed hashicorp/external v2.3.1
> 
> Terraform has been successfully initialized!
> 
> You may now begin working with Terraform. Try running "terraform plan" to see
> any changes that are required for your infrastructure. All Terraform commands
> should now work.
> 
> If you ever set or change modules or backend configuration for Terraform,
> rerun this command to reinitialize your working directory. If you forget, other
> commands will detect it and remind you to do so if necessary.
> ```
> 
> Would it be more reasonable to `exit` as update the code like this:
> 
> ```
> * ) 
>     echo "invalid input"
>     exit 0
>     ;;
> ```
> 
> Therefore, for the case of non y and non n case is better addressed.

Hi @kencho51,

You are right, but I think you are not testing the most up-to-date version of my branch (or you tested before my latest push), as the current latest commits has already this  implemented.

Reply:
@kencho51 
for some weird reasons the change mentioned in 

https://github.com/gigascience/gigadb-website/blob/6be9e0399258c2c2fef310d8688faaf7681f697e/ops/scripts/tf_init.sh

don't match my local git log which show your version.
Let me try fix that

Reply:
Don't know what happen but commit [1d98992](https://github.com/gigascience/gigadb-website/pull/1472/commits/1d989926e90224bb3c1824a422f2d86a32901578) below is restoring my original code

Reply:
Hi @rija,

The commit [1d98992](https://github.com/gigascience/gigadb-website/pull/1472/commits/1d989926e90224bb3c1824a422f2d86a32901578) provides great feature as if unknown option was supplied to the `tf_init.sh`, like:
```
% ../../../scripts/tf_init.sh --project gigascience/forks/kencho-gigadb-website --env staging --abcsd=12355
Invalid option: --abcsd=12355 in
../../../scripts/tf_init.sh --project gigascience/forks/kencho-gigadb-website --env staging --abcsd=12355
```

But my previous comment is when the user provides not y or not n choice in the step of `Do you want to continue (y/n)?`. Currently, the `tf_init.sh` would continue to execute if, like, `apple` was given:
```
% ../../../scripts/tf_init.sh --project gigascience/forks/kencho-gigadb-website --env staging 
Current directory: /Volumes/kencho/gigadb-website/ops/infrastructure/envs/staging
Project: gigascience/forks/kencho-gigadb-website
Environment: staging
Region: ap-northeast-2
GitLab User: kencho
Web EC2 Type: t3.micro
Bastion EC2 Type: t3.micro
RDS EC2 Type: t3.micro

Do you want to continue (y/n)?apple
invalid

Initializing the backend...
Initializing modules...

Initializing provider plugins...
- Reusing previous version of hashicorp/aws from the dependency lock file
- Reusing previous version of hashicorp/random from the dependency lock file
- Reusing previous version of hashicorp/external from the dependency lock file
- Using previously-installed hashicorp/aws v5.5.0
- Using previously-installed hashicorp/random v3.5.1
- Using previously-installed hashicorp/external v2.3.1

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```

And I was suggesting to exit the terraform init process when not y or not n was the input of the question.